nixos/modules/programs/spacefm.nix

```diff
@@ -0,0 +1,41 @@
+# Global configuration for spacefm.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.programs.spacefm;
+
+in
+{
+  ###### interface
+
+  options = {
+
+    programs.spacefm = {
+
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        example = literalExample ''{
```

Comment:
I'm concerned it will create a system-wide dependency on pkg.bash and pkg.gksu since all modules are installed. The reason I used literalExample and _''${pkgs.bash}_ was precisely to avoid this...

Still, it's a good idea... I'll look it up in other modules to see if anything like that was done. I suspect the .enable Boolean might be how it's circumvented but I'll have to make sure.